ingest-app/src/cmr/ingest/validation/validation.clj

```diff
@@ -98,35 +98,35 @@
 
 (defn- related-url-validator
   "Return a validator that checks a ContentType, Type, and Subtype keyword combo
-   plus Format and MimeType in GetData for Related URL field which can be inside
-   a ContactInformation or be a standalone field. ContactInformation can themselves
-   be found in DataCenters, ContactGroups, and ContactPersons."
-  [kms-index]
-  {:RelatedUrls 
-   [(match-related-url-kms-keywords-validations kms-index)
-    (v/every {:GetData {:MimeType (match-kms-keywords-validation-single
-                                   kms-index
-                                   :mime-type
-                                   msg/mime-type-not-matches-kms-keywords)}})]})
+   if warn? is set to false, otherwise will check MimeType in GetData for Related URL
+   field. All of which can be inside a ContactInformation or be a standalone field.
+   ContactInformation can themselves be found in DataCenters, ContactGroups, and ContactPersons."
+  [kms-index warn?]
```

Comment:
I don't know about this, I would think that this function would always return a validator and the issue is if it should be called or not, but I would defer to @sxu123 as she has done the last refactoring of this file. I am leaning in the direction that this is wrong.

Reply:
to my understanding the first part of `related-url-validator` should be returned as an error but the second part of the function should be returned as a warning, so would a better way be to create a separate function for the warning portion?

Reply:
I think so, but @sxu123 had the most opinions here last time, reach out to her and ask for advice. She reformatted my code and I think it is more readable now.

Reply:
Using another function instead of using flag would be more readable.